esrally/metrics.py

```diff
@@ -1366,21 +1353,13 @@ def to_result_dicts(self):
 
     @classmethod
     def from_dict(cls, d):
-        # for backwards compatibility with Rally < 0.9.2
```

Comment:
I've decided to include this cleanup here as well as this seemed to be too small to warrant a dedicated PR.